[Bug]: SyntaxError: Expected identifier but found "0"

### Describe the bug

Hi there!

`@module-federation/vite` 1.13.1 introduced a change in the way the entrypoint is built, causing a syntax error in the resulting code.

Here is an excerpt of the compiled entrypoint, shortened and formatted for readability:

```js
const __vite__mapDeps=(i,m=__vite__mapDeps,d=(m.f||(m.f=["./index-DJ82wtI2.css"])))=>i.map(i=>d[i]);
import{_ as a}from"./preload-helper-PPVm8Dsz.js";
const n={
  "@jahia/moonstone":async()=>await import("./index-H0OhSNsz.js"),__vite__mapDeps([0]),import.meta.url),
  "@jahia/ui-extender":async()=>await a(()=>import("./index-4uEGXMfF.js"),
  //...
}
```

The syntax error is one line 4, parentheses are not matched

According to my investigation, the bug appeared between [1.13.0 and 1.13.1](https://github.com/module-federation/vite/compare/1.13.0...1.13.1), but it's hard to pinpoint an exact commit.

Here is a reproduction:

```bash
gh repo clone tleclere-jahia/site-settings-languages
cd site-settings-languages
git checkout 5a7a36787690ee4ed814da7bf8ed7dc16cce4a73
yarn install # Requires yarn 4
yarn build

# Display the broken file, try to parse it with esbuild
cat src/main/resources/javascript/apps/assets/localSharedImportMap-*.js
npx esbuild --bundle src/main/resources/javascript/apps/index.js --format=esm --outfile=/dev/null
```

The repro is running Vite 7.3.1 and the lockfile is committed, you can check all versions used

Note: the [`@jahia/vite-federation-plugin`](https://github.com/Jahia/javascript-components/blob/master/packages/vite-federation-plugin/src/index.ts) is just a proxy on top of `@module-federation/vite` with common remotes configured and a custom runtime plugin

CC @tleclere-jahia

### Version

v7.3.1

### Reproduction

See above

### Relevant log output

```shell
✘ [ERROR] Expected identifier but found "0"

    src/main/resources/javascript/apps/assets/localSharedImportMap-C1wvUUYs.js:2:139:
      2 │ ...s";const n={"@jahia/moonstone":async()=>await import("./index-Blf7sSMR.js"),__vite__mapDeps([0]),import.meta.url),"@jahia/ui-extender":async()=>await a(()=>import("./index-DFnCNZUZ.js"),i...
        ╵                                                                                                 ^
```

### Validations

- [x] Read the [docs](https://github.com/module-federation/vite).
- [x] Read the [common issues list](https://github.com/module-federation/vite/issues).
- [x] Check that there isn't [already an issue](https://github.com/module-federation/vite/issues) that reports the same bug to avoid creating a duplicate.
- [x] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/src/utils/__tests__/controlChunkSanitizer.test.ts
+++ b/src/utils/__tests__/controlChunkSanitizer.test.ts
@@ -26,6 +26,39 @@ describe('controlChunkSanitizer', () => {
     ).toBe('import "./other.js";export const usedShared = {};');
   });
 
+  it('preserves preload helpers with non-empty dependency arrays', () => {
+    const code =
+      'import{_ as o}from"./preload-helper-BDBacUwf.js";' +
+      'const n={' +
+      '"@byte/api":async()=>await import("./index-DaqjAZdf.js"),' +
+      '"@byte/ui":async()=>await o(()=>import("./index-Bc0YS1wt.js"),__vite__mapDeps([0]),import.meta.url),' +
+      '"@byte/user-session":async()=>await o(()=>import("./index-BV4s8wZv.js"),[],import.meta.url),' +
+      '"react":async()=>await import("./index-DlZQ-_sN.js")' +
+      '}';
+
+    const result = stripEmptyPreloadCalls(code);
+
+    expect(result).toContain(
+      '"@byte/ui":async()=>await o(()=>import("./index-Bc0YS1wt.js"),__vite__mapDeps([0]),import.meta.url)'
+    );
+
+    expect(result).toContain('"@byte/user-session":async()=>await import("./index-BV4s8wZv.js")');
+
+    expect(result).not.toMatch(/await import\([^)]+\),__vite__mapDeps/);
+  });
+
+  it('does not break when only non-empty preload helpers exist', () => {
+    const code =
+      'import{_ as o}from"./preload-helper.js";' +
+      'const n={' +
+      '"@byte/ui":async()=>await o(()=>import("./ui.js"),__vite__mapDeps([0]),import.meta.url)' +
+      '}';
+
+    const result = stripEmptyPreloadCalls(code);
+
+    expect(result).toContain('o(()=>import("./ui.js"),__vite__mapDeps([0]),import.meta.url)');
+  });
+
   it('detects federation control chunks', () => {
     expect(isFederationControlChunk('remoteEntry.js', 'remoteEntry.js')).toBe(true);
     expect(isFederationControlChunk('assets/hostInit-abc.js', 'remoteEntry.js')).toBe(true);

--- a/src/utils/controlChunkSanitizer.ts
+++ b/src/utils/controlChunkSanitizer.ts
@@ -22,15 +22,20 @@ export function stripEmptyPreloadCalls(code: string): string {
       while (cursor < nextCode.length) {
         const char = nextCode[cursor];
         if (char === '(') depth++;
-        else if (char === ')') depth--;
-        else if (depth === 0 && nextCode.startsWith(',[],import.meta.url)', cursor)) {
+        else if (char === ')') {
+          depth--;
+          if (depth < 0) break;
+        } else if (depth === 0 && nextCode.startsWith(',[],import.meta.url)', cursor)) {
           replacementEnd = cursor;
           break;
         }
         cursor++;
       }
 
-      if (replacementEnd === -1) break;
+      if (replacementEnd === -1) {
+        start = nextCode.indexOf(marker, start + marker.length);
+        continue;
+      }
 
       const expression = nextCode.slice(exprStart, replacementEnd);
       nextCode =

